datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java

```diff
@@ -384,12 +390,68 @@ public SdxCluster getByNameInAccount(String userCrn, String name) {
         } catch (TransactionExecutionException e) {
             throw new TransactionRuntimeExecutionException(e);
         }
-
         FlowIdentifier flowIdentifier = sdxReactorFlowManager.triggerSdxCreation(savedSdxCluster);
-
         return Pair.of(savedSdxCluster, flowIdentifier);
     }
 
+    public Pair<SdxCluster, FlowIdentifier> resizeSdx(final String userCrn, final String clusterName, final SdxClusterResizeRequest sdxClusterResizeRequest) {
```

Comment:
this method has a lot in common with `create` method, is it not possible to create a common method for these two?

Reply:
There are similarities in initializing certain fields but overall there are a lot of diferences.

Reply:
i see, then at least you can extract some similar parts into methods, like these:
```
        SdxCluster sdxCluster = new SdxCluster();
        sdxCluster.setInitiatorUserCrn(userCrn);
        sdxCluster.setCrn(createCrn(getAccountIdFromCrn(userCrn)));
        sdxCluster.setClusterName(name);
        sdxCluster.setAccountId(getAccountIdFromCrn(userCrn));
        sdxCluster.setClusterShape(sdxClusterRequest.getClusterShape());
        sdxCluster.setCreated(clock.getCurrentTimeMillis());
        sdxCluster.setEnvName(environment.getName());
        sdxCluster.setEnvCrn(environment.getCrn());
        sdxCluster.setRangerRazEnabled(sdxClusterRequest.isEnableRangerRaz());
        sdxCluster.setSdxClusterServiceVersion(sdxClusterServiceVersion);
```

```
        try {
            newSdxCluster.setStackRequest(JsonUtil.writeValueAsString(stackRequest));
        } catch (JsonProcessingException e) {
            LOGGER.error("Can not parse internal stackrequest", e);
            throw new BadRequestException("Can not parse internal stackrequest", e);
        }
```

and there are several method calls which is used in both methods, for example:
`prepareDefaultSecurityConfigs`
`MDCBuilder.buildMdcContext(sdxCluster)`
`validateEnv`
`validateMediumDutySdxEnablement`
`prepareCloudStorageForStack`
`stackRequest.setResourceCrn(sdxCluster.getCrn())`

I know there is lot of variables and method calls here, but I am sure it is possible to avoid some duplication in this case.

Reply:
agree, there are lot of common things so we can put them into common methods, maybe we can move out these from SdxService to another service.

Reply:
will update.